gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallContext.java

```diff
@@ -535,7 +563,9 @@ public Channel getChannel() {
 
   /** The {@link CallOptions} set on this context. */
   public CallOptions getCallOptions() {
-    return callOptions;
+    if (!isDirectPath) return callOptions;
+    // Remove the CallCredentials attached to the callOptions if it's DirectPath.
+    return callOptions.withCallCredentials(null);
```

Comment:
Can this logic reside inside the constructor? I think it would be better for the getter to just return value.

nit: Can the comment be updated to reflect the `why`. Perhaps something like (probably needs better wording): `CallCredentials is stripped from CallOptions because CallCredentials are attached to ChannelCredentials in DirectPath flows. Adding it again would duplicate the headers.`

Reply:
Done. I also modify the merge logic to respect the isDirectPath of the coming context. Also several cases I take care of here:

1. If `withChannel` is subsequently called, we attach a CallCreds from the creds back to calloptions and revert `isDirectPath` to false.
2. If `withCallOptions` is subsequently called, we strip the CallCreds from it if DirectPath.
3. If `withCredentials` is subsequently called, we attach its corresponding CallCreds to the calloptions only if non-DirectPath

Please help check if I missed any corner case.